app/supplejack/extraction/execution.rb

```diff
@@ -48,7 +50,29 @@ def execution_cancelled?
     end
 
     def stop_condition_met?
```

Comment:
I think instead of doing this you can put the logging of each condition within the method itself, then you don't need to hang onto the conditions or have the logic in here. 

So calling it as it is now

```
    def stop_condition_met?
      [set_number_reached?, extraction_failed?, duplicate_document_extracted?, custom_stop_conditions_met?].any?(true)
    end
```

then doing something like this for each condition

```
    def set_number_reached?
      return false unless @harvest_job.present? && @harvest_job.pipeline_job.set_number?
      
      log_stop_condition_hit('set_number_reached', 'Set number limit reached', { condition_type: 'set_number_reached' })

      @harvest_job.pipeline_job.pages == @extraction_definition.page
    end
```